### PR TITLE
Fix format date time pattern

### PR DESCRIPTION
**Pull request by FrankYFTang**

### CODE REVIEW

**Comment by FrankYFTang** [comment]
Address the  FormatDateTimePattern issue found by @anba in https://github.com/tc39/proposal-intl-extend-timezonename/issues/5#issuecomment-834314743

**Comment by FrankYFTang** [comment]
@anba

**Comment by FrankYFTang** [comment]
I also propose a PR to fix ECMA402 in https://github.com/tc39/ecma402/pull/573

**Review by anba — APPROVED** [review]
LGTM with issues mentioned in tc39/ecma402#573 fixed. Thanks! :smile:

File: datetimeformat.html [hunk 1]
```diff
@@ -151,6 +151,11 @@ <h1>FormatDateTimePattern ( _dateTimeFormat_, _patternParts_, _x_, _rangeFormatO
             1. Let _f_ be the value of _dateTimeFormat_'s internal slot whose name is the Internal Slot column of the matching row.
             1. Let _fv_ be a String value representing the day period of _tm_ in the form given by _f_; the String value depends upon the implementation and the effective locale of _dateTimeFormat_.
             1. Append a new Record { [[Type]]: _p_, [[Value]]: _fv_ } as the last element of the list _result_.
+          1. <ins>Else if _p_ is equal to *"timeZoneName"*, then</ins>
+            1. <ins>Let _f_ be _dateTimeFormat_[[TimeZoneName]].</ins>
+            1. <ins>Let _v_ be _dateTimeFormat_[[TimeZone]].</ins>
+            1. <ins>Let _fv_ be a String value representing _v_ in the form given by _f_; the String value depends upon the implementation and the effective locale and calendar of _dateTimeFormat_. The String value may also depend on the value of the [[InDST]] field of _tm_ if _f_ is *"short"*, *"long"*, *"shortOffset"*, or *"longOffset"*.  If the implementation does not have a localized representation of _f_, then use the String value of _v_ itself.</ins>
```
Comment (ptomato) [inline datetimeformat.html]:
```suggestion
            1. <ins>Let _fv_ be a String value representing _v_ in the form given by _f_; the String value depends upon the implementation and the effective locale of _dateTimeFormat_. The String value may also depend on the value of the [[InDST]] field of _tm_ if _f_ is *"short"*, *"long"*, *"shortOffset"*, or *"longOffset"*.  If the implementation does not have a localized representation of _f_, then use the String value of _v_ itself.</ins>
```

File: datetimeformat.html [hunk 2]
```diff
@@ -151,6 +151,11 @@ <h1>FormatDateTimePattern ( _dateTimeFormat_, _patternParts_, _x_, _rangeFormatO
             1. Let _f_ be the value of _dateTimeFormat_'s internal slot whose name is the Internal Slot column of the matching row.
             1. Let _fv_ be a String value representing the day period of _tm_ in the form given by _f_; the String value depends upon the implementation and the effective locale of _dateTimeFormat_.
             1. Append a new Record { [[Type]]: _p_, [[Value]]: _fv_ } as the last element of the list _result_.
+          1. <ins>Else if _p_ is equal to *"timeZoneName"*, then</ins>
+            1. <ins>Let _f_ be _dateTimeFormat_[[TimeZoneName]].</ins>
```
Comment (ptomato) [inline datetimeformat.html]:
```suggestion
            1. <ins>Let _f_ be _dateTimeFormat_.[[TimeZoneName]].</ins>
```

File: datetimeformat.html [hunk 3]
```diff
@@ -151,6 +151,11 @@ <h1>FormatDateTimePattern ( _dateTimeFormat_, _patternParts_, _x_, _rangeFormatO
...
+            1. <ins>Let _v_ be _dateTimeFormat_[[TimeZone]].</ins>
```
Comment (ptomato) [inline datetimeformat.html]:
```suggestion
            1. <ins>Let _v_ be _dateTimeFormat_.[[TimeZone]].</ins>
```

**Review by ptomato — APPROVED** [review]
Seems correct. I marked some of the same comments that Ms2ger and anba left on https://github.com/tc39/ecma402/pull/573

Would it make sense to wait for that PR to land in the 402 spec text, and then "rebase" this one on top? That would mean that slightly different text would be marked with `<ins>` here.

**Pull request merged**